components/bsa/compressedbsafile.cpp

```diff
@@ -35,7 +35,16 @@
 
 #include <boost/iostreams/filtering_streambuf.hpp>
 #include <boost/iostreams/copy.hpp>
-#include <boost/iostreams/filter/zlib.hpp>
+
+#if defined(_WIN32)
+    #pragma warning (push)
+    #pragma warning (disable : 4706)
+    #include <boost/iostreams/filter/zlib.hpp>
+    #pragma warning (pop)
+#else
+    #include <boost/iostreams/filter/zlib.hpp>
+#endif
+
```

Comment:
Would (or has?) this ever get fixed upstream in boost?

Reply:
It [is not](https://github.com/boostorg/iostreams/blob/master/include/boost/iostreams/filter/zlib.hpp#L392) fixed. Maybe it is worth to create bugreport on the Boost repo about it.

Reply:
A [ticket](https://github.com/boostorg/iostreams/issues/131) on the Boost repo.

Reply:
Thanks!